op-challenger/types/game_type_test.go

```diff
@@ -0,0 +1,90 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestDefaultGameType returns the default dispute game type.
+func TestDefaultGameType(t *testing.T) {
+	require.Equal(t, AttestationDisputeGameType, DefaultGameType())
+}
+
+// TestGameType_Valid tests the Valid function with valid inputs.
+func TestGameType_Valid(t *testing.T) {
+	require.True(t, AttestationDisputeGameType.Valid())
+	require.True(t, FaultDisputeGameType.Valid())
+	require.True(t, ValidityDisputeGameType.Valid())
+}
+
+// TestGameType_Invalid tests the Valid function with an invalid input.
+func TestGameType_Invalid(t *testing.T) {
+	require.False(t, GameType(ValidityDisputeGameType+1).Valid())
+}
+
+// FuzzGameType_Invalid checks that invalid game types are correctly
+// returned as invalid by the validation [Valid] function.
+func FuzzGameType_Invalid(f *testing.F) {
+	maxCount := len(DisputeGameTypes)
+	f.Fuzz(func(t *testing.T, number uint8) {
+		if number >= uint8(maxCount) {
+			require.False(t, GameType(number).Valid())
+		} else {
+			require.True(t, GameType(number).Valid())
+		}
+	})
+}
+
+// TestGameType_Default tests the default value of the DisputeGameType.
+func TestGameType_Default(t *testing.T) {
+	d := NewDisputeGameType()
+	require.Equal(t, DefaultGameType(), d.selected)
+	require.Equal(t, DefaultGameType(), d.Type())
+}
+
+// TestGameType_String tests the Set and String function on the DisputeGameType.
+func TestGameType_String(t *testing.T) {
```

Comment:
This just seems to be testing the values against itself. If we want to test this, I recommend testing against hardcoded values.

Reply:
Yeah I'd say we want a list of the expected GameType->String pairs in this test.  This string test would confirm we get the expected values and `TestGameType_Invalid` would get the highest valid `GameType` value from that list. `TestGameType_Valid` could also then be a table test that iterates over the expected game types and checks they're all valid.

It does introduce some duplication because the game types are declared in two places but the production code version separates the definition of GameType from its string name which introduces the risk of them lining up, whereas the test version would have them paired together very clearly so its obvious in a diff if they aren't lining up properly.

Reply:
Actually, the other possibility here is that `GameType` should just be a string and would be its own name which avoids any chance of them being out of sync and simplifies a whole bunch of stuff.  Not sure if there's a particular reason we need it to be a uint8 though.